src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,192 @@
+        newUser.setAge(AGE_18);
+        newUser.setPassword(USER_PASSWORD_OK);
+    }
+
+    @Test
+    void register_nullUser_notOk() {
+        user = null;
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+            throw new RegistrationException(" ");
+        });
+    }
+
+    @Test
+    void register_emptyLogin_notOk() {
+        user.setLogin(" ");
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+            throw new RegistrationException("not allowed empty line!!!");
+        });
+    }
+
+    @Test
+    void register_nullLogin_notOk() {
+        user.setLogin(null);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+            throw new RegistrationException("empty line is not ok, please pass any login");
+        });
+    }
+
+    @Test
+    void register_sameLogin_notOk() {
+        newUser.setLogin(USER_NAME);
+        assertEquals(newUser.getLogin(), user.getLogin());
+    }
+
+    @Test
+    void register_rightLogin_ok() {
+        user.setLogin(NEW_USER_NAME);
+        assertNotEquals(newUser.getLogin(), user.getLogin());
+    }
+
+    @Test
+    void register_nullAge_notOk() {
+        user.setAge(null);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void register_negativeAge_notOk() {
+        user.setAge(NEGATIVE_AGE);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);;
+        });
+    }
+
+    @Test
+    void register_underAge_notOk() {
+        user.setAge(AGE_UNDER_18);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void register_overAge_notOk() {
+        user.setAge(OVER_MAX_AGE);
+        boolean overage = user.getAge() > MAX_AGE;
+        assertTrue(overage);
+    }
+
+    @Test
+    void register_maxAge_ok() {
+        user.setAge(MAX_AGE);
+        boolean maxAgeOk = user.getAge() <= MAX_AGE;
+        assertTrue(maxAgeOk);
+    }
+
+    @Test
+    void register_rightAge_ok() {
+        user.setAge(AGE_OVER_18);
+        boolean ageIsOk = user.getAge() > AGE_18;
+        assertTrue(ageIsOk);
+    }
+
+    @Test
+    void register_minAge_age() {
+        boolean minRequiredAgeOK = user.getAge() == AGE_18;
+        assertTrue(minRequiredAgeOK);
+    }
+
+    @Test
+    void register_nullPassword_notOk() {
+        user.setPassword(null);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void register_emptyPassword_notOk() {
+        user.setPassword(" ");
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });

```

Comment:
add this test case to `register_shortPasswordLength_notOk` and delete this method